gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/CleanupController.java

```diff
@@ -20,9 +20,37 @@ public CleanupController(CleanupService cleanupService) {
   }
 
   @ApiOperation(value = "Opt out of clean up for a marked resource.", response = HashMap.class)
-  @RequestMapping(method = RequestMethod.PUT, value = "/resources/{namespace}/{resourceId}/optOut")
-  Map optOut(@PathVariable String namespace,
+  @RequestMapping(method = RequestMethod.PUT,
+                  value = "/resources/{namespace}/{resourceId}/optOut",
+                  produces= "text/html")
+  String optOut(@PathVariable String namespace,
              @PathVariable String resourceId) {
-    return cleanupService.optOut(namespace, resourceId);
+    Map markedResource = cleanupService.optOut(namespace, resourceId);
+    if (markedResource.isEmpty()) {
+      return getHtmlWithMessage(namespace, resourceId, "Resource does not exist. Please try a valid resource.");
+    }
+
+    return getHtmlWithMessage(namespace, resourceId, "Resource has been restored, and opted out of future deletion!");
+  }
+
+  @ApiOperation(value = "Restore a resource that has been soft deleted.", response = HashMap.class)
+  @RequestMapping(method = RequestMethod.PUT,
+                  value = "/resources/{namespace}/{resourceId}/restore",
+                  produces= "text/html")
+  String restore(@PathVariable String namespace,
+             @PathVariable String resourceId) {
+    String status = cleanupService.restore(namespace, resourceId);
+    if (status.equals("404")) {
+      return getHtmlWithMessage(namespace, resourceId, "Resource does not exist. Please try a valid resource.");
+    }
+
+    return getHtmlWithMessage(namespace, resourceId, "Resource has been opted out!");
+  }
+
+  private String getHtmlWithMessage(String namespace, String resourceId, String message) {
+    return "<body>\n" +
+      "Swabbie Resource: [namespace=" + namespace + ", resourceId=" + resourceId + "] \n" +
```

Comment:
HTML treats `\n` as just whitespace.  Did you want this to show up on a new line?
```suggestion
      "Swabbie Resource: [namespace=" + namespace + ", resourceId=" + resourceId + "] <br>\n" +
```

Reply:
yes! Thank you.